connman/gweb/gweb.c

```diff
@@ -475,6 +477,30 @@ static inline void call_result_func(struct web_session *session, guint16 status)
 
 }
 
+static inline int call_dns_route_func(struct web_session *session, bool add)
+{
+	bool ret;
+
+	if (add == session->dns_routes_set)
+		return -EALREADY;
```

Comment:
I'm wondering about return values from this function

Nothing to do, everything is already as requested -> EALREADY *error* -> sounds like every caller should be prepared to handle this special case - which most do not which then makes one wonder whether it is intentional or not.

Reply:
Well, I find it useful to differentiate between ok return and nothing done return. This is a pattern in many places around ConnMan to return `EALREADY` for a no-op when states match. Adding handling to callers for this.

Reply:
This [a4e3c22](https://github.com/sailfishos/connman/pull/55/commits/a4e3c2202bab7ec34a107eac78bab9c12b6567ff) should handle the issue here.